sources/Renderer/Direct3D11/Shader/D3D11Shader.cpp

```diff
@@ -156,17 +156,22 @@ bool D3D11Shader::CompileSource(ID3D11Device* device, const ShaderDescriptor& sh
     std::string fileContent;
     const char* sourceCode      = nullptr;
     SIZE_T      sourceLength    = 0;
+    const char* sourceName      = nullptr;
 
     if (shaderDesc.sourceType == ShaderSourceType::CodeFile)
     {
         fileContent     = ReadFileString(shaderDesc.source);
         sourceCode      = fileContent.c_str();
         sourceLength    = fileContent.size();
+
+        if (shaderDesc.source != nullptr)
```

Comment:
I think it would be good to always prefer `shaderDesc.name` if it's specified and only use `shaderDesc.source` as a fallback case. The rest looks good to me.